rpc/wrtc_signaling_answerer.go

```diff
@@ -145,7 +151,9 @@ func (ans *webrtcSignalingAnswerer) startAnswerer() {
 		}
 		return answerClient, nil
 	}
-	ans.activeBackgroundWorkers.Add(1)
+	ans.answerMu.RLock()
```

Comment:
> This patch right now doesn't have that strong invariant. Which makes proving the patch, as-is, doesn't have a bug much more "fun" than it needs to be.

An excellent description of the issue here hahaha. You're totally right that this patch is not enforcing as _strong_ of an invariant as the last; let me add a context and see what you think.

Reply:
> Because I claim we'll always hit the client == nil check on L160. Proving that is left to the reader.

heh, still working through this one... but thank you for outlining the bug we are vulnerable to if `client == nil` did not save us. 

I agree that implementing the invariant you describe would make it easier to reason that there is no such bad flow. I'm also okay leaving as is if implementing said invariant leads to significant rework.

Reply:
I _think_ I enforced the stronger variant with the context checks I added.

Reply:
Yep, I agree the code succeeded in doing that.